2013-10-31  Hang Yeh  <yehh@lnxsrv02.seas.ucla.edu>

	* ls.c (cmp_mtime): I changed this function to put separate cases for ti
        me that is before 1970 and after since the negative value is affecting t        he results. 

--- ls.c	2013-10-30 14:32:56.468746000 -0700
+++ patch_l.c	2013-10-30 14:31:57.749030000 -0700
@@ -2875,22 +2875,12 @@
 
 static inline int
 cmp_mtime (struct fileinfo const *a, struct fileinfo const *b,
-           int (*cmp) (char const *, char const *))
+	   int (*cmp) (char const *, char const *))
 {
-  int diff;
-  if (b->stat.st_mtim.tv_sec < 0 && a->stat.st_mtim.tv_sec < 0)
-        { diff = timespec_cmp (get_stat_mtime (&b->stat),
-                             get_stat_mtime (&a->stat)); }
-  else if (b->stat.st_mtim.tv_sec < 0)
-        { diff = -1;        }
-  else if (a->stat.st_mtim.tv_sec < 0)
-        { diff = 1; }
-  else
-        { diff = timespec_cmp (get_stat_mtime (&b->stat),
-                                  get_stat_mtime (&a->stat));}
-   return diff ? diff : cmp (a->name, b->name);
+  int diff = timespec_cmp (get_stat_mtime (&b->stat),
+			   get_stat_mtime (&a->stat));
+  return diff ? diff : cmp (a->name, b->name);
 }
-
 static int compare_mtime (V a, V b) { return cmp_mtime (a, b, xstrcoll); }
 static int compstr_mtime (V a, V b) { return cmp_mtime (a, b, strcmp); }
 static int rev_cmp_mtime (V a, V b) { return compare_mtime (b, a); }
